feature(report): Templates V2 - Elastic abstaction

oxypomme opened the pull request:
# Report

> [!IMPORTANT]  
> Old templates (template v1) support is dropped, please use `ezreeport-admin migrate <data-folder>` to transform your templates to the new format

## Features

- Template V2
 - Simplified aggregation management and other figures params
 - Template V1 support is dropped, please use `ezreeport-admin migrate <data-folder>` to transform your templates to the new format

## Bug fixs

- More details with errors while rendering 
- Fixed various timezone issues

# ezreeport-admin

Added migration

oxypomme committed:
feat(report): simplified aggregation management and other figures params

BREAKING CHANGE: tasks and templates are now in v2 and v1 support is dropped

oxypomme committed:
fix(ezra): fixed issue with logs on import

oxypomme committed:
feat(ezra): added migration script for templates v2 (2.0.0)

oxypomme committed:
feat(report): re-added support for counts

oxypomme committed:
fix(report): fixed issue with new render

oxypomme committed:
fix(report): added more details when fetch error happen

oxypomme committed:
feat(report): sort data before rendering

oxypomme committed:
fix(ezra): fixed order beign in aggregations

oxypomme committed:
fix(report): fixed issue with bars with no colors

oxypomme committed:
fix(report): fixed date detection

oxypomme committed:
fix(report): fixed metrics when values are 0

oxypomme committed:
fix(report): fixed issue with colormap and non string values

oxypomme committed:
fix(ezra): fixed migration not getting metric aggregations

oxypomme committed:
fix(report): using TZ for cron configs

oxypomme committed:
fix(report): fixed imprecision of count

oxypomme committed:
fix(report): fixed various timezone issues

felixleo22 commented on services/report/src/models/reports/fetch/index.ts:
```diff
@@ -0,0 +1,99 @@
+    username: Type.String({ minLength: 1 }),
+  }),
+  dateField: Type.String({ minLength: 1 }),
+  index: Type.Optional(Type.String({ minLength: 1 })),
```
Not optional

felixleo22 commented on services/report/src/models/reports/fetch/index.ts:
```diff
@@ -0,0 +1,99 @@
+      options.period,
+    );
+
+    // TODO: add cause in error
```
Bro ?

felixleo22 commented on services/report/src/models/reports/fetch/results.ts:
```diff
@@ -0,0 +1,220 @@
+          value: aggregation.value ?? 0,
+        };
+      },
+    ).filter((result) => !!result);
```
Remove .filter((result) => !!result);

felixleo22 commented on services/report/src/models/reports/fetch/results.ts:
```diff
@@ -0,0 +1,220 @@
+): FetchResultItem[] {
+  checkEsErrors(response, errorCause);
+
+  if (!('aggregations' in response) || !response.aggregations) {
```
Move it in checkEsErrors

felixleo22 commented on services/report/src/app.ts:
```diff
@@ -18,6 +19,9 @@ const { port, allowedOrigins: rawOrigins } = config;
 const start = async () => {
   // Create Fastify instance
   const fastify = Fastify({
+    ajv: {
```
Check https://github.com/ezpaarse-project/readlabs/blob/main/services/api/src/app.ts#L54

felixleo22 commented on services/report/src/models/reports/render.ts:
```diff
@@ -27,6 +27,7 @@ import { Type, type Static, assertIsSchema } from '~/lib/typebox';
 
 import type { FigureType } from '~/models/figures';
 import { Layout } from '~/models/layouts';
+import { FetchResultItem } from './fetch/results';
```
import type

felixleo22 commented on services/report/src/lib/pdf/metrics.ts:
```diff
@@ -105,14 +100,45 @@ const formatNumber = (
+    }
+  } catch (error) {
+    if (!(error instanceof Error)) {
+      throw new Error(`An unexpected error occurred wile formatting "${label.text}" ("${value}"): ${error}`);
```
wile

oxypomme committed:
fix(report): minor fixs and doc on new fetch methods

oxypomme committed:
refactor(report): fastify now use global ajv instance

oxypomme committed:
fix(report): fixed issues when creating a task from a preset

oxypomme committed:
feat(sdk): updated templates types, following api

BREAKING CHANGE: might need to review how you create templates

oxypomme committed:
feat(sdk): deleted deprecated methods

BREAKING CHANGE: start/stop crons, pause/resume crons, update task (not upsert), update template (not upsert), history (now activity)

oxypomme committed:
feat(report): deleted deprecated routes

BREAKING CHANGE: start/stop crons, pause/resume crons, update task (not upsert), update template (not upsert)

oxypomme committed:
feat(report): added targets in history

oxypomme committed:
feat(report): added more details on how targets are unsubscribed

oxypomme committed:
fix(sdk): fixed issue in template types

oxypomme committed:
Merge branch 'dev' into feature/templates-v2
